README.md

```diff
@@ -0,0 +1,241 @@
+In OpenFeature, we refer to this as [targeting](https://openfeature.dev/specification/glossary#targeting).
+If the flag management system you're using supports targeting, you can provide the input data using the [evaluation context](https://openfeature.dev/docs/reference/concepts/evaluation-context).
+
+```python
+from openfeature.api import (
+    get_client,
+    get_provider,
+    set_provider
+    get_evaluation_context,
+    set_evaluation_context,
+)
+
+global_context = EvaluationContext(
+    targeting_key="targeting_key1", attributes={"application": "value1"}
+)
+request_context = EvaluationContext(
+    targeting_key="targeting_key2", attributes={"email": request.form['email']}
+)
+
+## set global context
+set_evaluation_context(first_context)
+
+# merge second context
+client = get_client(name="No-op Provider", version="0.5.2")
+client.get_string_value("email", None, request_context)
+
+```
+
+### Hooks
+
+[Hooks](https://openfeature.dev/docs/reference/concepts/hooks) allow for custom logic to be added at well-defined points of the flag evaluation life-cycle.
+Look [here](https://openfeature.dev/ecosystem/?instant_search%5BrefinementList%5D%5Btype%5D%5B0%5D=Hook&instant_search%5BrefinementList%5D%5Btechnology%5D%5B0%5D=python) for a complete list of available hooks.
+If the hook you're looking for hasn't been created yet, see the [develop a hook](#develop-a-hook) section to learn how to build it yourself.
+
+Once you've added a hook as a dependency, it can be registered at the global, client, or flag invocation level.
+
+```python
+# set global hooks at the API-level
+from openfeature.api import add_hooks
+add_hooks([MyHook()])
+
+# or configure them in the client
+client = OpenFeatureClient()
+client.add_hooks([MyHook()])
+```
+
+<!-- TODO: example of invocation-level hook -->
+
+### Logging
+
+Logging customization is not yet available in the Python SDK.
+
+### Named clients
+
+Named clients are not yet available in the Python SDK. Progress on this feature can be tracked [here](https://github.com/open-feature/python-sdk/issues/125).
+
+### Eventing
+
+Events are not yet available in the Python SDK. Progress on this feature can be tracked [here](https://github.com/open-feature/python-sdk/issues/125).
+
+### Shutdown
+
+A shutdown method is not yet available in the Python SDK. Progress on this feature can be tracked [here](https://github.com/open-feature/python-sdk/issues/125).
+
+## Extending
+
+### Develop a provider
+
+To develop a provider, you need to create a new project and include the OpenFeature SDK as a dependency.
+This can be a new repository or included in [the existing contrib repository](https://github.com/open-feature/python-sdk-contrib) available under the OpenFeature organization.
+You’ll then need to write the provider by implementing the `FeatureProvider` interface exported by the OpenFeature SDK.
```

Comment:
It's called `AbstractProvider` in the SDK, though I have been wondering if we should rename it for consistency with other SDKs.

Reply:
I'm not big on enforcing specific class/method/function/member names. I think that's something that the implementation should have a lot of flexibility on.

For example, I understand there isn't formal `Interfaces` in Python, due to multiple inheritance etc... so if an abstract base class is the right choice, and `Abstract` is a common prefix for such classes, then `AbstractProvider` makes perfect sense. **_I think the important thing to consider is what naming will communicate the core _ideas_ of the SDK to a Python developer most clearly_** - I trust your ability to answer that question much more than mine.

Reply:
I've updated the comment based on everyones feedback.